### ISSUE TITLE
bind volume on WSL2 uses resolved path

### ISSUE THREAD

**Issue by Xiot** [docker/for-win#7431]
I am creating a bind volume into my docker that points to the WSL2 share `/mnt/d/htpc/downloads`.
```
  downloads:
    name: downloads
    driver: local
    driver_opts:
      type: none
      device: /mnt/d/htpc/downloads
      o: bind
```

The volume gets created successfully, and the container can use it just fine, however it appears to resolve the path to the location in the WSL2 container when it saves it.  This is the inspect for the volume
```
    {
        "CreatedAt": "2020-06-25T21:38:50Z",
        "Driver": "local",
        "Labels": {
            "com.docker.compose.project": "htpc",
            "com.docker.compose.version": "1.25.5",
            "com.docker.compose.volume": "downloads"
        },
        "Mountpoint": "/var/lib/docker/volumes/downloads/_data",
        "Name": "downloads",
        "Options": {
            "device": "/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/aae79361fc23fb5016071bdaa6d2d524b68cb712568e8989f1577aa71296a289",
            "o": "bind",
            "type": "none"
        },
        "Scope": "local"
    }
```
The issue comes in if I have to run `docker-compose` again, as it complains that the volume doesn't have the same device, and asks me to remove and re-create it.

  - [ ] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID: 25EEA6A6-98AE-4F51-9201-E1FCE8DFBE76/20200625214029

### Expected behavior
Create a volume for `/mnt/d/htpc/downloads` and allow for successive `docker-compose` calls without having to rebind the volume.

### Actual behavior
After the first `docker-compose`, everything looks good.
The second one however gives me this.
```
 Configuration for volume downloads specifies "device" driver_opt /mnt/d/htpc/downloads, but a volume with the same name uses a different "device" driver_opt (/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/aae79361fc23fb5016071bdaa6d2d524b68cb712568e8989f1577aa71296a289). If you wish to use the new configuration, please remove the existing volume "downloads" first:
```

### Information
Please, help us understand the problem.  For instance:
  - Is it reproducible?  - YES
  - Is the problem new?  - I only started playing with Docker a couple days ago.
  - Did the problem appear with an update? NO
  - A reproducible case if this is a bug, Dockerfiles with reproduction inside is best.

  - Windows Version: Windows 10 v 2004 - Build 19041.329
  - Docker Desktop Version: 2.3.0.3
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: NO

### Steps to reproduce the behavior
```
version: "3.7"
services:
    testbind:
        image: alpine
        container_name: testbind
        volumes:
            - downloads:/downloads
volumes:
    downloads:
        name: downloads
        driver: local
        driver_opts:
          type: none
          device: ${DOWNLOADS_PATH}
          o: bind
```
```
docker-compose up
docker-compose up
```

**Comment by ThomasQvist** [comment]
I am experiencing exactly the same behaviour

**Comment by ThomasQvist** [comment]
Any news on this issue (which really is a deal-breaker for me regarding docker on WSL2).
Any workarounds for having a persistent volume on Docker for WSL2?

Thank you for any help :-)

**Comment by simonferquel** [comment]
The issue is tracked in our backlog, we'll start working on it soon

**Comment by cakoops** [comment]
@simonferquel 
Hi Simon, any estimate when a fix for this issue is available?

I experience this problem both with volumes specified in the docker-compose yml file as well as with external volumes created with a docker volume create command. Should i open a second use for external volumes or is that not necessary.

**Comment by georgy-komarov** [comment]
Is there any activity on this issue?

I'm having the same problem with relative path
My workdir is ```C:\Users\User\PycharmProjects\courses-backend\courses```
I'm also having good start for the first time, however the second one fails

Here is the error message
```
ERROR: Configuration for volume courses-data specifies "device" driver_opt C:\Users\User\PycharmProjects\courses-backend\courses\data, but a volume with the same name uses a different "device" driver_opt (/run/desktop/mnt/host/c/Us
ers/User/PycharmProjects/courses-backend/courses/data). 
If you wish to use the new configuration, please remove the existing volume "courses_courses-data" first:
$ docker volume rm courses_courses-data
```

Here is the part of my docker-compose.yml
```
volumes:
  courses-data:
    driver: local
    driver_opts:
      type: none
      device: ./data
      o: bind
```

**Comment by simonferquel** [comment]
Edge build 2.4.2.0 (which will be availble later today) solves the issue. If feedback is good, it should get available on stable on early november.

**Comment by schniper** [comment]
I have just tested this on the edge build and it seems to behave the same. I am using ~/Sites as the device.
It works fine the first time I start the containers, but after a reboot it complains about checking paths.
Inspecting the volume, I have noticed that after the reboot, the ~/Sites got rewritten to /home/my_user/Sites. If I remove the volume and re-execute the docker-compose config, it works again until the next reboot.
Consequent docker-compose downs and ups without a reboot work fine, so I guess something happens when docker stops.
